retry.go

```diff
@@ -41,7 +42,10 @@ const DefaultRetryCap = time.Second * 30
 
 // newRetryTimer creates a timer with exponentially increasing
 // delays until the maximum retry attempts are reached.
-func (c Client) newRetryTimer(maxRetry int, unit time.Duration, cap time.Duration, jitter float64, doneCh chan struct{}) <-chan int {
+func (c Client) newRetryTimer(ctx context.Context, maxRetry int, unit time.Duration, cap time.Duration, jitter float64) <-chan int {
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
This needs to be done in executeMethod() instead.

Because we need executeMethod() to tell the retry timer to stop when executeMethod is going to stop